test/e2e/operator_test.go

```diff
@@ -0,0 +1,276 @@
+		t.Fatalf("ClusterOperator never became ready: %v", err)
+	}
+
+	// Ensure the VolumeSnapshot CRD is installed
+	_, err = client.ApiextensionsV1beta1Interface.CustomResourceDefinitions().Get("volumesnapshots.snapshot.storage.k8s.io", metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Error attempting to retrieve snapshot CRD: %v", err)
+	}
+
+	// Create the PV
+	pv := CreatePV(driver.GetDriverInfo().Name, "pv", size)
+	pv, err = client.CoreV1Interface.PersistentVolumes().Create(pv)
+	if err != nil {
+		t.Fatalf("Error attempting to create PV: %v", err)
+	}
+	defer func() {
+		err = client.CoreV1Interface.PersistentVolumes().Delete(pv.ObjectMeta.Name, &metav1.DeleteOptions{})
+		if err != nil {
+			t.Errorf("Error attempting to delete PV: %v", err)
+		}
+	}()
+
+	// Create the PVC
+	pvc := CreatePVC(namespace, pv.Spec.StorageClassName, pv.ObjectMeta.UID, size)
+	pvc, err = client.CoreV1Interface.PersistentVolumeClaims(namespace).Create(pvc)
+	if err != nil {
+		t.Fatalf("Error attempting to create PVC: %v", err)
+	}
+	defer func() {
+		err = client.CoreV1Interface.PersistentVolumeClaims(namespace).Delete(pvc.ObjectMeta.Name, &metav1.DeleteOptions{})
+		if err != nil {
+			t.Errorf("Error attempting to delete PVC: %v", err)
+		}
+	}()
+
+	// Create the VolumeSnapshotClass
+	snapshotClass := CreateFakeSnapshotClass(driver.GetDriverInfo().Name)
+	snapshotClass, err = client.SnapshotV1beta1Interface.VolumeSnapshotClasses().Create(snapshotClass)
+	if err != nil {
+		t.Fatalf("Error attempting to create VolumeSnapshotClass: %v", err)
+	}
+	defer func() {
+		err = client.SnapshotV1beta1Interface.VolumeSnapshotClasses().Delete(snapshotClass.ObjectMeta.Name, &metav1.DeleteOptions{})
+		if err != nil {
+			t.Errorf("Error attempting to delete VolumeSnapshotClass: %v", err)
+		}
+	}()
+	t.Logf("Created snapshotClass: %v", snapshotClass.ObjectMeta.Name)
+
+	// Create the VolumeSnapshot
+	snapshot := CreateFakeSnapshot(pvc.ObjectMeta.Name, namespace, snapshotClass.ObjectMeta.Name, pvc.ObjectMeta.UID)
+	snapshot, err = client.SnapshotV1beta1Interface.VolumeSnapshots(namespace).Create(snapshot)
+	if err != nil {
+		t.Fatalf("Error attempting to create VolumeSnapshot: %v", err)
+	}
+	defer func() {
+		err = client.SnapshotV1beta1Interface.VolumeSnapshots(namespace).Delete(snapshot.ObjectMeta.Name, &metav1.DeleteOptions{})
+		if err != nil {
+			t.Errorf("Error attempting to delete VolumeSnapshot: %v", err)
+		}
+	}()
+	t.Logf("Created snapshot %v", snapshot.ObjectMeta.Name)
+
+	// Search for a matching VolumeSnapshotContent
+	snapshotContent, err := waitForSnapshotContent(t, client, snapshot)
+	if err != nil {
+		t.Fatalf("Unable to find a matching VolumeSnapshotContent within the expected time: %v", err)
+	}
+	// Clean up the VolumeSnapshotContent
+	defer func() {
+		err = deleteSnapshotContent(t, client, snapshotContent)
+		if err != nil {
+			t.Logf("Error: %v", err)
+		}
+	}()
+	t.Logf("Found snapshot content %v", snapshotContent.ObjectMeta.Name)
+
```

Comment:
It would be better to have explicit `markSnapshotContentReady()` here instead of hiding it in `waitForSnapshotContent`

Reply:
I've updated `setStatus()` to be `markSnapshotContentReady()` and moved this to the main test.